### ISSUE TITLE
buildweb.sh - npm warnings and failure

### ISSUE THREAD

**Issue by iilei** [anthonynsimon/parrot#26]
Hi,

Testing the install locally, I came across a bunch of NPM warnings;

````
Installing dependencies, this might take a few minutes...
npm WARN @angular/core@2.4.5 requires a peer of rxjs@^5.0.1 but none was installed.
npm WARN @angular/core@2.4.5 requires a peer of zone.js@^0.7.2 but none was installed.
npm WARN @angular/http@2.4.5 requires a peer of rxjs@^5.0.1 but none was installed.
npm WARN @angular/core@2.4.5 requires a peer of zone.js@^0.7.2 but none was installed.
npm WARN @angular/router@3.4.5 requires a peer of rxjs@^5.0.1 but none was installed.
npm WARN @angular/core@2.4.5 requires a peer of zone.js@^0.7.2 but none was installed.
npm WARN @ngtools/webpack@1.2.4 requires a peer of webpack@2.2.0 but none was installed.
npm WARN angular2-jwt@0.1.28 requires a peer of rxjs@^5.0.0 but none was installed.
npm WARN @angular/core@2.4.5 requires a peer of zone.js@^0.7.2 but none was installed.
npm WARN codelyzer@1.0.0-beta.4 requires a peer of @angular/compiler@~2.2.0 but none was installed.
npm WARN codelyzer@1.0.0-beta.4 requires a peer of @angular/core@~2.2.0 but none was installed.
npm WARN extract-text-webpack-plugin@2.0.0-rc.0 requires a peer of webpack@^2.2.0 but none was installed.
````

and despite emitting a bunch of files, it exits with code 1, so the dist files won`t get copied over to nginx public folder.

Remarkable Line of Error output (amongst a bunch of errors of the same kind):

```
...
ERROR in ./~/flag-icon-css/css/flag-icon.min.css
Module build failed: Error: Cannot find module '/Users/iilei/codebase/parrot/web-app/node_modules/css-loader/index.js?sourcemap&minimize'
...
```

*ng -v*
angular-cli: 1.0.0-beta.20-4
node: 6.9.0

I tried to solve the peer dependency issues by hand with no avail.

Would you recommend another node version, or can you provide a shrinkwrap file (or yarn.lock) to pin the versions?

Thanks a bunch!

Cheers

**Comment by anthonynsimon** [comment]
Hey!

I was taking a look, I also get the npm peer/dependency warnings.

It seems to be a known issue of angular-cli: [install of angular-cli results in unmeet peer dependencies. Installing a peer does not resolve #1941](https://github.com/angular/angular-cli/issues/1941)

Also here [angular 2.1.0 release causes missing peer dependencies #2692](https://github.com/angular/angular-cli/issues/2692)

I then tried by installing angular-cli globally (ng) and running it with sudo (`sudo ng build -prod`), it then worked as usual.

The scripts work properly this way. If you clone the master branch and run the following, it should work:

```
$ sudo bash scripts/buildweb.sh
$ sudo bash scripts/start.sh
```

Let me know if that makes it work. I know it's not the best to have to install angular-cli globally, and give root permissions, hopefully they sort that out.

**Comment by iilei** [comment]
Thank you 😃 
Yes, running as root works. However, running it as non-root as with PR #23 is what I consider a preliminary step before I am going to work on my own fork and hopefully be able to make my own contributions via PRs, too 😉

**Comment by iilei** [comment]
This Issue would be resolved once #29 is merged :neckbeard:

**Comment by anthonynsimon** [comment]
Just tried it out, that's awesome!
